API.md

```diff
@@ -0,0 +1,337 @@
+> and keep using it to initialize the KeysManager every time you restart.
+> This key_seed is used to derive your node's secret key (which corresponds to its node pubkey) and all other secret key material.
+
+I was wondering, can't LDK take of this?
+It can be hooked with a persister, 
+which could potentially persist it and restore it on restart.
+The comment above seems to indicate that it's the user's responsibility to write it to disk and restore it.
+
+
+## Fresh Start vs Restart
+
+While reading [the guide](https://lightningdevkit.org/docs/build_node/),
+I noticed that in several places the user has to write different code depending on whether doing a fresh start or a restart.
+I feel it would really help the user if there was no distinction at all,
+and LDK would take care of differentiation between the two situations.
+
+My previous comment on key seed management is motivated by this.
+
+
+## Channel identifiers
+
+Channels have ids associated with them, represented by the `OutPoint` class.
+As noted in the [persist section in the build node guide](https://lightningdevkit.org/docs/build_node#initialize-persist),
+you'd expect to write the channel data keyed by instances of `OutPoint`.
+
+However, `OutPoint` does not override `equals` nor `hashCode`,
+hence it does not take the contents of `OutPoint` into account to see if two instances are the same.
+By default, two instances are equals if they point to the same object,
+not if the contents are the same.
+For anything you want as key in a map or dictionary, 
+you want the latter property.
+Whenever you override `equals`, 
+also override `hashCode` to avoid surprises in structures such as hash maps.
+Unless you're using a Java version that includes records,
+I normally let the IDE (e.g. IntelliJ) generate them for me.
+They allow you to select which fields to use for computing `equals`/`hashCode`.
+More details [here](https://www.baeldung.com/java-equals-hashcode-contracts).
+
+
+## Finalizers
+
+I've noticed that some classes, e.g. `OutPut` rely on Java finalizers.
+Finalizers in Java are generally considered a bad idea.
+[This article](https://www.baeldung.com/java-finalize) explains the pitfalls in detail.
+
+The best practice for resource management in Java these days,
+is to make your classes extend `AutoCloseable`, 
+which then enables you to pass them into a try-with-resources construct.
+See [detailed explanation](https://www.baeldung.com/java-try-with-resources).
+
+
+## NioPeerHandler
+
+Assuming the current status quo (using Java NIO directly),
+how does someone close the sockets that `NioPeerHandler` uses?
+As per the previous section, 
+the general consensus is that it should extend `AutoCloseable`,
+and close any socket or other resources it has created in the `close` method implementation.
+
+
+## Style
+
+The JDK and majority of Java libraries use camel case style throughout their source.
```

Comment:
Do you think its worth auto-converting all the _[a-z] to [A-Z]? I don't think that would be hard.